INFO: Scrapy 1.6.0 started (bot: scrapybot)
INFO: Versions: lxml 4.3.3.0, libxml2 2.9.9, cssselect 1.0.3, parsel 1.5.1, w3lib 1.20.0, Twisted 19.7.0, Python 3.7.2 (v3.7.2:9a3ffc0492, Dec 24 2018, 02:44:43) - [Clang 6.0 (clang-600.0.57)], pyOpenSSL 18.0.0 (OpenSSL 1.1.0j  20 Nov 2018), cryptography 2.4.2, Platform Darwin-18.7.0-x86_64-i386-64bit
INFO: Overridden settings: {'AUTOTHROTTLE_ENABLED': True, 'AUTOTHROTTLE_MAX_DELAY': 3, 'AUTOTHROTTLE_START_DELAY': 1, 'FEED_FORMAT': 'csv', 'FEED_URI': 'medium_full_2009-2011.csv', 'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'}
INFO: Telnet Password: fa54ab969e6786db
INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.feedexport.FeedExporter',
 'scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.throttle.AutoThrottle']
INFO: Enabled downloader middlewares:
['scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware',
 'scrapy.downloadermiddlewares.downloadtimeout.DownloadTimeoutMiddleware',
 'scrapy.downloadermiddlewares.defaultheaders.DefaultHeadersMiddleware',
 'scrapy.downloadermiddlewares.useragent.UserAgentMiddleware',
 'scrapy.downloadermiddlewares.retry.RetryMiddleware',
 'scrapy.downloadermiddlewares.redirect.MetaRefreshMiddleware',
 'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware',
 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware',
 'scrapy.downloadermiddlewares.cookies.CookiesMiddleware',
 'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware',
 'scrapy.downloadermiddlewares.stats.DownloaderStats']
INFO: Enabled spider middlewares:
['scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
INFO: Enabled item pipelines:
[]
INFO: Spider opened
INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
INFO: Telnet console listening on 127.0.0.1:6023
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2009> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0-----------------------> from <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0----------------------->
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2010> (referer: None)
DEBUG: Redirecting (302) to <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=b16d42baaf16> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0----------------------->
DEBUG: Redirecting (302) to <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=8bdc53de2dd8> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2011> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
ERROR: Spider error processing <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=8bdc53de2dd8> (referer: https://medium.com/tag/data-science/archive/2009)
DEBUG: Crawled (200) <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=b16d42baaf16> (referer: https://medium.com/tag/data-science/archive/2009)
ERROR: Spider error processing <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=8bdc53de2dd8> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
ERROR: Spider error processing <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=b16d42baaf16> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=b18b6ea41cb2> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=fc6d0b293a09> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=b18b6ea41cb2> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=b18b6ea41cb2> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=fc6d0b293a09> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=fc6d0b293a09> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
INFO: Closing spider (finished)
INFO: Dumping Scrapy stats:
{'downloader/request_bytes': 8256,
 'downloader/request_count': 17,
 'downloader/request_method_count/GET': 17,
 'downloader/response_bytes': 277941,
 'downloader/response_count': 17,
 'downloader/response_status_count/200': 9,
 'downloader/response_status_count/302': 8,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2019, 11, 21, 3, 44, 5, 161246),
 'log_count/DEBUG': 17,
 'log_count/ERROR': 6,
 'log_count/INFO': 9,
 'memusage/max': 48668672,
 'memusage/startup': 48668672,
 'request_depth_max': 1,
 'response_received_count': 9,
 'scheduler/dequeued': 17,
 'scheduler/dequeued/memory': 17,
 'scheduler/enqueued': 17,
 'scheduler/enqueued/memory': 17,
 'spider_exceptions/KeyError': 6,
 'start_time': datetime.datetime(2019, 11, 21, 3, 43, 54, 554708)}
INFO: Spider closed (finished)
INFO: Scrapy 1.6.0 started (bot: scrapybot)
INFO: Versions: lxml 4.3.3.0, libxml2 2.9.9, cssselect 1.0.3, parsel 1.5.1, w3lib 1.20.0, Twisted 19.7.0, Python 3.7.2 (v3.7.2:9a3ffc0492, Dec 24 2018, 02:44:43) - [Clang 6.0 (clang-600.0.57)], pyOpenSSL 18.0.0 (OpenSSL 1.1.0j  20 Nov 2018), cryptography 2.4.2, Platform Darwin-18.7.0-x86_64-i386-64bit
INFO: Overridden settings: {'AUTOTHROTTLE_ENABLED': True, 'AUTOTHROTTLE_MAX_DELAY': 3, 'AUTOTHROTTLE_START_DELAY': 1, 'FEED_FORMAT': 'csv', 'FEED_URI': 'medium_full_2009-2011.csv', 'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'}
INFO: Telnet Password: 98ccc08be0a109f2
INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.feedexport.FeedExporter',
 'scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.throttle.AutoThrottle']
INFO: Enabled downloader middlewares:
['scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware',
 'scrapy.downloadermiddlewares.downloadtimeout.DownloadTimeoutMiddleware',
 'scrapy.downloadermiddlewares.defaultheaders.DefaultHeadersMiddleware',
 'scrapy.downloadermiddlewares.useragent.UserAgentMiddleware',
 'scrapy.downloadermiddlewares.retry.RetryMiddleware',
 'scrapy.downloadermiddlewares.redirect.MetaRefreshMiddleware',
 'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware',
 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware',
 'scrapy.downloadermiddlewares.cookies.CookiesMiddleware',
 'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware',
 'scrapy.downloadermiddlewares.stats.DownloaderStats']
INFO: Enabled spider middlewares:
['scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
INFO: Enabled item pipelines:
[]
INFO: Spider opened
INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
INFO: Telnet console listening on 127.0.0.1:6023
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2009> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1----------------------->
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0-----------------------> from <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2010> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=2623ab2fecea> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Redirecting (302) to <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=a93245f01e3c> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2011> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
ERROR: Spider error processing <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=a93245f01e3c> (referer: https://medium.com/tag/data-science/archive/2009)
DEBUG: Crawled (200) <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=2623ab2fecea> (referer: https://medium.com/tag/data-science/archive/2009)
ERROR: Spider error processing <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=a93245f01e3c> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
ERROR: Spider error processing <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=2623ab2fecea> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=c3dc3401c63f> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2----------------------->
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=6ab22cae3725> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=c3dc3401c63f> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=c3dc3401c63f> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=6ab22cae3725> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=6ab22cae3725> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
INFO: Closing spider (finished)
INFO: Dumping Scrapy stats:
{'downloader/request_bytes': 8256,
 'downloader/request_count': 17,
 'downloader/request_method_count/GET': 17,
 'downloader/response_bytes': 278311,
 'downloader/response_count': 17,
 'downloader/response_status_count/200': 9,
 'downloader/response_status_count/302': 8,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2019, 11, 21, 3, 45, 10, 85160),
 'log_count/DEBUG': 17,
 'log_count/ERROR': 6,
 'log_count/INFO': 9,
 'memusage/max': 45408256,
 'memusage/startup': 45408256,
 'request_depth_max': 1,
 'response_received_count': 9,
 'scheduler/dequeued': 17,
 'scheduler/dequeued/memory': 17,
 'scheduler/enqueued': 17,
 'scheduler/enqueued/memory': 17,
 'spider_exceptions/KeyError': 6,
 'start_time': datetime.datetime(2019, 11, 21, 3, 44, 59, 367955)}
INFO: Spider closed (finished)
INFO: Scrapy 1.6.0 started (bot: scrapybot)
INFO: Versions: lxml 4.3.3.0, libxml2 2.9.9, cssselect 1.0.3, parsel 1.5.1, w3lib 1.20.0, Twisted 19.7.0, Python 3.7.2 (v3.7.2:9a3ffc0492, Dec 24 2018, 02:44:43) - [Clang 6.0 (clang-600.0.57)], pyOpenSSL 18.0.0 (OpenSSL 1.1.0j  20 Nov 2018), cryptography 2.4.2, Platform Darwin-18.7.0-x86_64-i386-64bit
INFO: Overridden settings: {'AUTOTHROTTLE_ENABLED': True, 'AUTOTHROTTLE_MAX_DELAY': 3, 'AUTOTHROTTLE_START_DELAY': 1, 'FEED_FORMAT': 'csv', 'FEED_URI': 'medium_full_2009-2011.csv', 'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'}
INFO: Telnet Password: 52ff4f44dcdd7743
INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.feedexport.FeedExporter',
 'scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.throttle.AutoThrottle']
INFO: Enabled downloader middlewares:
['scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware',
 'scrapy.downloadermiddlewares.downloadtimeout.DownloadTimeoutMiddleware',
 'scrapy.downloadermiddlewares.defaultheaders.DefaultHeadersMiddleware',
 'scrapy.downloadermiddlewares.useragent.UserAgentMiddleware',
 'scrapy.downloadermiddlewares.retry.RetryMiddleware',
 'scrapy.downloadermiddlewares.redirect.MetaRefreshMiddleware',
 'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware',
 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware',
 'scrapy.downloadermiddlewares.cookies.CookiesMiddleware',
 'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware',
 'scrapy.downloadermiddlewares.stats.DownloaderStats']
INFO: Enabled spider middlewares:
['scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
INFO: Enabled item pipelines:
[]
INFO: Spider opened
INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
INFO: Telnet console listening on 127.0.0.1:6023
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2009> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0-----------------------> from <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0----------------------->
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2010> (referer: None)
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2011> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=c232885ff634> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Redirecting (302) to <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=4408cc38ed96> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0----------------------->
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
ERROR: Spider error processing <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=4408cc38ed96> (referer: https://medium.com/tag/data-science/archive/2009)
ERROR: Spider error processing <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=4408cc38ed96> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=c232885ff634> (referer: https://medium.com/tag/data-science/archive/2009)
ERROR: Spider error processing <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=c232885ff634> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=489ad180d61d> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=98c39b0da01> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=489ad180d61d> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=489ad180d61d> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=98c39b0da01> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=98c39b0da01> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
INFO: Closing spider (finished)
INFO: Dumping Scrapy stats:
{'downloader/request_bytes': 8255,
 'downloader/request_count': 17,
 'downloader/request_method_count/GET': 17,
 'downloader/response_bytes': 278309,
 'downloader/response_count': 17,
 'downloader/response_status_count/200': 9,
 'downloader/response_status_count/302': 8,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2019, 11, 21, 3, 53, 17, 419091),
 'log_count/DEBUG': 17,
 'log_count/ERROR': 6,
 'log_count/INFO': 9,
 'memusage/max': 45387776,
 'memusage/startup': 45387776,
 'request_depth_max': 1,
 'response_received_count': 9,
 'scheduler/dequeued': 17,
 'scheduler/dequeued/memory': 17,
 'scheduler/enqueued': 17,
 'scheduler/enqueued/memory': 17,
 'spider_exceptions/KeyError': 6,
 'start_time': datetime.datetime(2019, 11, 21, 3, 53, 7, 444179)}
INFO: Spider closed (finished)
INFO: Scrapy 1.6.0 started (bot: scrapybot)
INFO: Versions: lxml 4.3.3.0, libxml2 2.9.9, cssselect 1.0.3, parsel 1.5.1, w3lib 1.20.0, Twisted 19.7.0, Python 3.7.2 (v3.7.2:9a3ffc0492, Dec 24 2018, 02:44:43) - [Clang 6.0 (clang-600.0.57)], pyOpenSSL 18.0.0 (OpenSSL 1.1.0j  20 Nov 2018), cryptography 2.4.2, Platform Darwin-18.7.0-x86_64-i386-64bit
INFO: Overridden settings: {'AUTOTHROTTLE_ENABLED': True, 'AUTOTHROTTLE_MAX_DELAY': 3, 'AUTOTHROTTLE_START_DELAY': 1, 'FEED_FORMAT': 'csv', 'FEED_URI': 'medium_full_2009-2011.csv', 'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'}
INFO: Telnet Password: da68523f1b735979
INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.feedexport.FeedExporter',
 'scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.throttle.AutoThrottle']
INFO: Enabled downloader middlewares:
['scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware',
 'scrapy.downloadermiddlewares.downloadtimeout.DownloadTimeoutMiddleware',
 'scrapy.downloadermiddlewares.defaultheaders.DefaultHeadersMiddleware',
 'scrapy.downloadermiddlewares.useragent.UserAgentMiddleware',
 'scrapy.downloadermiddlewares.retry.RetryMiddleware',
 'scrapy.downloadermiddlewares.redirect.MetaRefreshMiddleware',
 'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware',
 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware',
 'scrapy.downloadermiddlewares.cookies.CookiesMiddleware',
 'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware',
 'scrapy.downloadermiddlewares.stats.DownloaderStats']
INFO: Enabled spider middlewares:
['scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
INFO: Enabled item pipelines:
[]
INFO: Spider opened
INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
INFO: Telnet console listening on 127.0.0.1:6023
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2009> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0-----------------------> from <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0----------------------->
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2010> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=55d8c15b850a> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2011> (referer: None)
DEBUG: Redirecting (302) to <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=aec0acb2ac39> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0----------------------->
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
ERROR: Spider error processing <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
DEBUG: Crawled (200) <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=aec0acb2ac39> (referer: https://medium.com/tag/data-science/archive/2009)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1----------------------->
ERROR: Spider error processing <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
ERROR: Spider error processing <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=aec0acb2ac39> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=55d8c15b850a> (referer: https://medium.com/tag/data-science/archive/2009)
ERROR: Spider error processing <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=55d8c15b850a> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
INFO: Received SIGINT, shutting down gracefully. Send again to force 
INFO: Closing spider (shutdown)
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=9ea0ea26e1db> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2----------------------->
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=e63292a2bafc> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1----------------------->
INFO: Dumping Scrapy stats:
{'downloader/request_bytes': 7436,
 'downloader/request_count': 15,
 'downloader/request_method_count/GET': 15,
 'downloader/response_bytes': 222020,
 'downloader/response_count': 15,
 'downloader/response_status_count/200': 7,
 'downloader/response_status_count/302': 8,
 'finish_reason': 'shutdown',
 'finish_time': datetime.datetime(2019, 11, 21, 3, 53, 35, 368854),
 'log_count/DEBUG': 15,
 'log_count/ERROR': 4,
 'log_count/INFO': 10,
 'memusage/max': 45395968,
 'memusage/startup': 45395968,
 'request_depth_max': 1,
 'response_received_count': 7,
 'scheduler/dequeued': 15,
 'scheduler/dequeued/memory': 15,
 'scheduler/enqueued': 17,
 'scheduler/enqueued/memory': 17,
 'spider_exceptions/KeyError': 4,
 'start_time': datetime.datetime(2019, 11, 21, 3, 53, 29, 938914)}
INFO: Spider closed (shutdown)
INFO: Scrapy 1.6.0 started (bot: scrapybot)
INFO: Versions: lxml 4.3.3.0, libxml2 2.9.9, cssselect 1.0.3, parsel 1.5.1, w3lib 1.20.0, Twisted 19.7.0, Python 3.7.2 (v3.7.2:9a3ffc0492, Dec 24 2018, 02:44:43) - [Clang 6.0 (clang-600.0.57)], pyOpenSSL 18.0.0 (OpenSSL 1.1.0j  20 Nov 2018), cryptography 2.4.2, Platform Darwin-18.7.0-x86_64-i386-64bit
INFO: Overridden settings: {'AUTOTHROTTLE_ENABLED': True, 'AUTOTHROTTLE_MAX_DELAY': 3, 'AUTOTHROTTLE_START_DELAY': 1, 'FEED_FORMAT': 'csv', 'FEED_URI': 'medium_full_2009-2011.csv', 'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'}
INFO: Telnet Password: 5c4ee2f6bdf5af4d
INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.feedexport.FeedExporter',
 'scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.throttle.AutoThrottle']
INFO: Enabled downloader middlewares:
['scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware',
 'scrapy.downloadermiddlewares.downloadtimeout.DownloadTimeoutMiddleware',
 'scrapy.downloadermiddlewares.defaultheaders.DefaultHeadersMiddleware',
 'scrapy.downloadermiddlewares.useragent.UserAgentMiddleware',
 'scrapy.downloadermiddlewares.retry.RetryMiddleware',
 'scrapy.downloadermiddlewares.redirect.MetaRefreshMiddleware',
 'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware',
 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware',
 'scrapy.downloadermiddlewares.cookies.CookiesMiddleware',
 'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware',
 'scrapy.downloadermiddlewares.stats.DownloaderStats']
INFO: Enabled spider middlewares:
['scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
INFO: Enabled item pipelines:
[]
INFO: Spider opened
INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
INFO: Telnet console listening on 127.0.0.1:6023
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2009> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0-----------------------> from <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0----------------------->
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2010> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=985c4b7a4006> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Redirecting (302) to <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=74a054c37340> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2011> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
INFO: Received SIGINT, shutting down gracefully. Send again to force 
INFO: Closing spider (shutdown)
ERROR: Spider error processing <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=985c4b7a4006> (referer: https://medium.com/tag/data-science/archive/2009)
ERROR: Spider error processing <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=985c4b7a4006> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=74a054c37340> (referer: https://medium.com/tag/data-science/archive/2009)
ERROR: Spider error processing <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=74a054c37340> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=7d225d2cdbc1> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
ERROR: Spider error processing <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
INFO: Dumping Scrapy stats:
{'downloader/request_bytes': 6727,
 'downloader/request_count': 14,
 'downloader/request_method_count/GET': 14,
 'downloader/response_bytes': 219031,
 'downloader/response_count': 14,
 'downloader/response_status_count/200': 7,
 'downloader/response_status_count/302': 7,
 'finish_reason': 'shutdown',
 'finish_time': datetime.datetime(2019, 11, 21, 3, 54, 10, 108867),
 'log_count/DEBUG': 14,
 'log_count/ERROR': 4,
 'log_count/INFO': 10,
 'memusage/max': 45473792,
 'memusage/startup': 45473792,
 'request_depth_max': 1,
 'response_received_count': 7,
 'scheduler/dequeued': 14,
 'scheduler/dequeued/memory': 14,
 'scheduler/enqueued': 16,
 'scheduler/enqueued/memory': 16,
 'spider_exceptions/KeyError': 4,
 'start_time': datetime.datetime(2019, 11, 21, 3, 54, 5, 384830)}
INFO: Spider closed (shutdown)
INFO: Scrapy 1.6.0 started (bot: scrapybot)
INFO: Versions: lxml 4.3.3.0, libxml2 2.9.9, cssselect 1.0.3, parsel 1.5.1, w3lib 1.20.0, Twisted 19.7.0, Python 3.7.2 (v3.7.2:9a3ffc0492, Dec 24 2018, 02:44:43) - [Clang 6.0 (clang-600.0.57)], pyOpenSSL 18.0.0 (OpenSSL 1.1.0j  20 Nov 2018), cryptography 2.4.2, Platform Darwin-18.7.0-x86_64-i386-64bit
INFO: Overridden settings: {'AUTOTHROTTLE_ENABLED': True, 'AUTOTHROTTLE_MAX_DELAY': 3, 'AUTOTHROTTLE_START_DELAY': 1, 'FEED_FORMAT': 'csv', 'FEED_URI': 'medium_full_2009-2011.csv', 'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'}
INFO: Telnet Password: 6dcbf97281c52992
INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.feedexport.FeedExporter',
 'scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.throttle.AutoThrottle']
INFO: Enabled downloader middlewares:
['scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware',
 'scrapy.downloadermiddlewares.downloadtimeout.DownloadTimeoutMiddleware',
 'scrapy.downloadermiddlewares.defaultheaders.DefaultHeadersMiddleware',
 'scrapy.downloadermiddlewares.useragent.UserAgentMiddleware',
 'scrapy.downloadermiddlewares.retry.RetryMiddleware',
 'scrapy.downloadermiddlewares.redirect.MetaRefreshMiddleware',
 'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware',
 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware',
 'scrapy.downloadermiddlewares.cookies.CookiesMiddleware',
 'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware',
 'scrapy.downloadermiddlewares.stats.DownloaderStats']
INFO: Enabled spider middlewares:
['scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
INFO: Enabled item pipelines:
[]
INFO: Spider opened
INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
INFO: Telnet console listening on 127.0.0.1:6023
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2009> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0-----------------------> from <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0----------------------->
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1----------------------->
DEBUG: Redirecting (302) to <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=12bc1fb89fb0> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2010> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=9f4a88a568a1> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2011> (referer: None)
INFO: Received SIGINT, shutting down gracefully. Send again to force 
INFO: Closing spider (shutdown)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=12bc1fb89fb0> (referer: https://medium.com/tag/data-science/archive/2009)
ERROR: Spider error processing <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=12bc1fb89fb0> (referer: https://medium.com/tag/data-science/archive/2009)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Crawled (200) <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
ERROR: Spider error processing <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
Traceback (most recent call last):
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 83, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-paddingBottom3 a').attrib['href']
KeyError: 'href'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/utils/defer.py", line 102, in iter_errback
    yield next(it)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/offsite.py", line 29, in process_spider_output
    for x in result:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/referer.py", line 339, in <genexpr>
    return (_set_referer(r) for r in result or ())
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/urllength.py", line 37, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/scrapy/spidermiddlewares/depth.py", line 58, in <genexpr>
    return (r for r in result or () if _filter(r))
  File "/Users/nancy/PycharmProjects/medium-ds-articles/src/scripts/full_article_data/full_article_data_2009-2011.py", line 85, in parse_tags
    item['linkOfAuthorProfile'] = response.css('div.u-noWrapWithEllipsis a').attrib['href']
KeyError: 'href'
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1----------------------->
INFO: Scrapy 1.6.0 started (bot: scrapybot)
INFO: Versions: lxml 4.3.3.0, libxml2 2.9.9, cssselect 1.0.3, parsel 1.5.1, w3lib 1.20.0, Twisted 19.7.0, Python 3.7.2 (v3.7.2:9a3ffc0492, Dec 24 2018, 02:44:43) - [Clang 6.0 (clang-600.0.57)], pyOpenSSL 18.0.0 (OpenSSL 1.1.0j  20 Nov 2018), cryptography 2.4.2, Platform Darwin-18.7.0-x86_64-i386-64bit
INFO: Overridden settings: {'AUTOTHROTTLE_ENABLED': True, 'AUTOTHROTTLE_MAX_DELAY': 3, 'AUTOTHROTTLE_START_DELAY': 1, 'FEED_FORMAT': 'csv', 'FEED_URI': 'medium_full_2009-2011.csv', 'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'}
INFO: Telnet Password: c1763fae34c1800c
INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.feedexport.FeedExporter',
 'scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.throttle.AutoThrottle']
INFO: Enabled downloader middlewares:
['scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware',
 'scrapy.downloadermiddlewares.downloadtimeout.DownloadTimeoutMiddleware',
 'scrapy.downloadermiddlewares.defaultheaders.DefaultHeadersMiddleware',
 'scrapy.downloadermiddlewares.useragent.UserAgentMiddleware',
 'scrapy.downloadermiddlewares.retry.RetryMiddleware',
 'scrapy.downloadermiddlewares.redirect.MetaRefreshMiddleware',
 'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware',
 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware',
 'scrapy.downloadermiddlewares.cookies.CookiesMiddleware',
 'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware',
 'scrapy.downloadermiddlewares.stats.DownloaderStats']
INFO: Enabled spider middlewares:
['scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
INFO: Enabled item pipelines:
[]
INFO: Spider opened
INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
INFO: Telnet console listening on 127.0.0.1:6023
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2009> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0-----------------------> from <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0----------------------->
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2010> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=a322f3749f49> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fmedium.quantiply.com%2Frethinking-cloud-enterprise-computing-8205851c2c02%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Redirecting (302) to <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=b59ba44282f1> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fdeveloperzen.com%2Fdata-mining-handling-missing-values-the-database-bd2241882e72%3Fsource%3Dtag_archive---------0----------------------->
DEBUG: Crawled (200) <GET https://medium.com/tag/data-science/archive/2011> (referer: None)
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=a322f3749f49> (referer: https://medium.com/tag/data-science/archive/2009)
DEBUG: Scraped from <200 https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=a322f3749f49>
{'NumOfClaps': [],
 'NumOfComments': None,
 'article': None,
 'articleTags': [],
 'articleURL': 'https://medium.quantiply.com/rethinking-cloud-enterprise-computing-8205851c2c02?source=tag_archive---------1-----------------------&gi=a322f3749f49',
 'nameOfAuthor': None,
 'postingTime': None,
 'readingTime': ' '}
DEBUG: Redirecting (302) to <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1-----------------------> from <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2010)
DEBUG: Scraped from <200 https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0----------------------->
{'NumOfClaps': [],
 'NumOfComments': None,
 'article': None,
 'articleTags': [],
 'articleURL': 'https://medium.com/stanfordreview/learning-from-last-year-part-ii-61b5c716e8ca?source=tag_archive---------0-----------------------',
 'nameOfAuthor': None,
 'postingTime': None,
 'readingTime': ' '}
DEBUG: Crawled (200) <GET https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=b59ba44282f1> (referer: https://medium.com/tag/data-science/archive/2009)
DEBUG: Scraped from <200 https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=b59ba44282f1>
{'NumOfClaps': [],
 'NumOfComments': None,
 'article': None,
 'articleTags': [],
 'articleURL': 'https://developerzen.com/data-mining-handling-missing-values-the-database-bd2241882e72?source=tag_archive---------0-----------------------&gi=b59ba44282f1',
 'nameOfAuthor': None,
 'postingTime': None,
 'readingTime': ' '}
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=eded54eba65a> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-1-ac97f950d6f8%3Fsource%3Dtag_archive---------2----------------------->
DEBUG: Crawled (200) <GET https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------> (referer: https://medium.com/tag/data-science/archive/2011)
DEBUG: Scraped from <200 https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0----------------------->
{'NumOfClaps': [],
 'NumOfComments': None,
 'article': None,
 'articleTags': [],
 'articleURL': 'https://medium.com/dark-matter-and-trojan-horses/of-brains-and-cities-neuroscience-and-cultures-of-decision-making-6bc6abb48d4b?source=tag_archive---------0-----------------------',
 'nameOfAuthor': None,
 'postingTime': None,
 'readingTime': ' '}
DEBUG: Redirecting (302) to <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=7e5fc66dfa5a> from <GET https://medium.com/m/global-identity?redirectUrl=https%3A%2F%2Fblog.matters.tech%2Fsearch-engine-vs-database-in-bi-part-2-1b191363424d%3Fsource%3Dtag_archive---------1----------------------->
DEBUG: Crawled (200) <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=eded54eba65a> (referer: https://medium.com/tag/data-science/archive/2011)
DEBUG: Scraped from <200 https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=eded54eba65a>
{'NumOfClaps': [],
 'NumOfComments': None,
 'article': None,
 'articleTags': [],
 'articleURL': 'https://blog.matters.tech/search-engine-vs-database-in-bi-part-1-ac97f950d6f8?source=tag_archive---------2-----------------------&gi=eded54eba65a',
 'nameOfAuthor': None,
 'postingTime': None,
 'readingTime': ' '}
DEBUG: Crawled (200) <GET https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=7e5fc66dfa5a> (referer: https://medium.com/tag/data-science/archive/2011)
DEBUG: Scraped from <200 https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=7e5fc66dfa5a>
{'NumOfClaps': [],
 'NumOfComments': None,
 'article': None,
 'articleTags': [],
 'articleURL': 'https://blog.matters.tech/search-engine-vs-database-in-bi-part-2-1b191363424d?source=tag_archive---------1-----------------------&gi=7e5fc66dfa5a',
 'nameOfAuthor': None,
 'postingTime': None,
 'readingTime': ' '}
INFO: Closing spider (finished)
INFO: Stored csv feed (6 items) in: medium_full_2009-2011.csv
INFO: Dumping Scrapy stats:
{'downloader/request_bytes': 8256,
 'downloader/request_count': 17,
 'downloader/request_method_count/GET': 17,
 'downloader/response_bytes': 277904,
 'downloader/response_count': 17,
 'downloader/response_status_count/200': 9,
 'downloader/response_status_count/302': 8,
 'finish_reason': 'finished',
 'finish_time': datetime.datetime(2019, 11, 21, 14, 1, 44, 701652),
 'item_scraped_count': 6,
 'log_count/DEBUG': 23,
 'log_count/INFO': 10,
 'memusage/max': 78008320,
 'memusage/startup': 78008320,
 'request_depth_max': 1,
 'response_received_count': 9,
 'scheduler/dequeued': 17,
 'scheduler/dequeued/memory': 17,
 'scheduler/enqueued': 17,
 'scheduler/enqueued/memory': 17,
 'start_time': datetime.datetime(2019, 11, 21, 14, 1, 35, 64146)}
INFO: Spider closed (finished)
INFO: Scrapy 1.6.0 started (bot: scrapybot)
INFO: Versions: lxml 4.3.3.0, libxml2 2.9.9, cssselect 1.0.3, parsel 1.5.1, w3lib 1.20.0, Twisted 19.7.0, Python 3.7.2 (v3.7.2:9a3ffc0492, Dec 24 2018, 02:44:43) - [Clang 6.0 (clang-600.0.57)], pyOpenSSL 18.0.0 (OpenSSL 1.1.0j  20 Nov 2018), cryptography 2.4.2, Platform Darwin-18.7.0-x86_64-i386-64bit
INFO: Overridden settings: {'AUTOTHROTTLE_ENABLED': True, 'AUTOTHROTTLE_MAX_DELAY': 3, 'AUTOTHROTTLE_START_DELAY': 1, 'FEED_FORMAT': 'csv', 'FEED_URI': '/Users/nancy/PycharmProjects/medium-ds-articles/data/raw/medium_titles_2012.csv', 'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)'}
INFO: Telnet Password: 9e3a91d6b06cff4b
INFO: Enabled extensions:
['scrapy.extensions.corestats.CoreStats',
 'scrapy.extensions.telnet.TelnetConsole',
 'scrapy.extensions.memusage.MemoryUsage',
 'scrapy.extensions.feedexport.FeedExporter',
 'scrapy.extensions.logstats.LogStats',
 'scrapy.extensions.throttle.AutoThrottle']
INFO: Enabled downloader middlewares:
['scrapy.downloadermiddlewares.httpauth.HttpAuthMiddleware',
 'scrapy.downloadermiddlewares.downloadtimeout.DownloadTimeoutMiddleware',
 'scrapy.downloadermiddlewares.defaultheaders.DefaultHeadersMiddleware',
 'scrapy.downloadermiddlewares.useragent.UserAgentMiddleware',
 'scrapy.downloadermiddlewares.retry.RetryMiddleware',
 'scrapy.downloadermiddlewares.redirect.MetaRefreshMiddleware',
 'scrapy.downloadermiddlewares.httpcompression.HttpCompressionMiddleware',
 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware',
 'scrapy.downloadermiddlewares.cookies.CookiesMiddleware',
 'scrapy.downloadermiddlewares.httpproxy.HttpProxyMiddleware',
 'scrapy.downloadermiddlewares.stats.DownloaderStats']
INFO: Enabled spider middlewares:
['scrapy.spidermiddlewares.httperror.HttpErrorMiddleware',
 'scrapy.spidermiddlewares.offsite.OffsiteMiddleware',
 'scrapy.spidermiddlewares.referer.RefererMiddleware',
 'scrapy.spidermiddlewares.urllength.UrlLengthMiddleware',
 'scrapy.spidermiddlewares.depth.DepthMiddleware']
INFO: Enabled item pipelines:
[]
INFO: Spider opened
INFO: Crawled 0 pages (at 0 pages/min), scraped 0 items (at 0 items/min)
INFO: Telnet console listening on 127.0.0.1:6023
